lldb/include/lldb/Target/PostMortemProcess.h

```diff
@@ -33,6 +34,23 @@ class PostMortemProcess : public Process {
   FileSpec GetCoreFile() const override { return m_core_file; }
 
 protected:
+  typedef lldb_private::Range<lldb::addr_t, lldb::addr_t> FileRange;
+  typedef lldb_private::RangeDataVector<lldb::addr_t, lldb::addr_t, FileRange>
+      VMRangeToFileOffset;
+  typedef lldb_private::RangeDataVector<lldb::addr_t, lldb::addr_t, uint32_t>
+      VMRangeToPermissions;
+
+  virtual const uint8_t *PeekMemory(lldb::addr_t low, lldb::addr_t high,
+                                    size_t &size) = 0;
```

Comment:
return `llvm::ArrayRef<uint8_t>` and get rid of the `size_t &available_bytes` parameter as the size of the returned ArrayRef will be sufficient. Do we want to return a `std::optional<llvm::ArrayRef<uint8_t>>` where if something comes back with no value, then we know PeekMemory is not implemented and we can fall back to the original code?

Reply:
It is sufficient to return an empty ArrayRef. Also, that is what all other APIs use in lldb. I will keep it the return type as llvm::ArrayRef<uint8_t> but let me know if you do want me to use std::optional.

Reply:
> It is sufficient to return an empty ArrayRef. Also, that is what all other APIs use in lldb. I will keep it the return type as llvm::ArrayRef<uint8_t> but let me know if you do want me to use std::optional.

If returning a empty `llvm::ArrayRef<uint8_t>` is enough for us to be able to fall back to `ReadMemory` then this is fine.